apps/backend/src/auth/auth.service.ts

```diff
@@ -11,20 +11,33 @@ export class AuthService {
     private jwtService: JwtService
   ) {}
 
-  async findOrCreateUser(userProfile: AuthSchProfile): Promise<User> {
-    const user = await this.prisma.user.findUnique({ where: { authSchId: userProfile.authSchId } });
-    if (user) return user;
-    return await this.prisma.user.create({
-      data: {
-        authSchId: userProfile.authSchId,
-        fullName: userProfile.displayName,
-        nickName: userProfile.firstName,
-        email: userProfile.email,
-        isSchResident: true,
-        //TODO: Find a solution for only dorm residents.
-        isActiveVikStudent: userProfile.bmeStatus.includes(BmeUnitScope.BME_VIK_ACTIVE),
-      },
-    });
+  async updateOrCreateUser(userProfile: AuthSchProfile): Promise<User> {
+    try {
+      return await this.prisma.user.update({
+        where: { authSchId: userProfile.authSchId },
+        data: {
+          isActiveVikStudent: userProfile.bme.bmeStatus.includes(BmeUnitScope.BME_VIK_ACTIVE),
+          vikStatusUpdatedAt: new Date(),
+        },
+      });
+    } catch (e) {
+      if (e instanceof Prisma.PrismaClientKnownRequestError) {
```

Comment:
is it elegant to do it in a try - catch?

Reply:
One less dB query than first finding the user, then updating it, so I think it's justified 

But I agree that from reading the code, it's not obvious when the user creation happens, but Prima's shitty error handling is to blame there xd

Reply:
fair, maybe the function could have a better name
what about findAndUpdateOrCreateUser?
or i would just keep it findOrCreateUser
but it is ofc not the most important question now i guess

Reply:
but is it (findAndUpdate)OrCreateUser or findAnd(UpdateOrCreate)User? Can we put ()-in function names? xdd